src/PieChart/components/PieChartContainer.ts

```diff
@@ -0,0 +1,205 @@
+import { PieChart } from "./PieChart";
+
+export type ChartType = "pie" | "donut";
+
+interface WrapperProps {
+    class?: string;
+    mxform: mxui.lib.form._FormBase;
+    mxObject?: mendix.lib.MxObject;
+    style?: string;
+    readOnly: boolean;
+}
+
+export interface PieChartContainerProps extends WrapperProps {
+    dataEntity: string;
+    dataSourceType: "xpath" | "microflow";
+    entityConstraint: string;
+    dataSourceMicroflow: string;
+    nameAttribute: string;
+    valueAttribute: string;
+    sortAttribute: string;
+    chartType: ChartType;
+    colorAttribute: string;
+    showToolBar: boolean;
+    showLegend: boolean;
+    responsive: boolean;
+    widthUnit: "percentage" | "pixels";
+    width: number;
+    heightUnit: "percentageOfWidth" | "pixels" | "percentageOfParent";
+    height: number;
+}
+
+interface PieChartContainerState {
+    alertMessage?: string;
+    colors?: string[];
+    labels?: string[];
+    values?: number[];
+}
+
+export default class PieChartContainer extends Component<PieChartContainerProps, PieChartContainerState> {
+    private subscriptionHandles: number[] = [];
+    private values: number[] = [];
+    private labels: string[] = [];
+    private colors: string[] = [];
+
+    constructor(props: PieChartContainerProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: PieChartContainer.validateProps(this.props),
+            colors: [],
+            labels: [],
+            values: []
+        };
+        this.fetchData = this.fetchData.bind(this);
+    }
+
+    render() {
+        if (this.props.mxObject) {
+            if (this.state.alertMessage) {
+                return createElement(Alert, {
+                    className: `widget-${this.props.chartType}-chart-alert`,
+                    message: this.state.alertMessage
+                });
+            }
+            return createElement(PieChart, {
+                className: this.props.class,
+                config: {
+                    displayModeBar: this.props.showToolBar
+                },
+                data: [ {
+                    hole: this.props.chartType === "donut" ? .4 : 0,
+                    hoverinfo: "label",
+                    labels: this.state.labels,
+                    marker: {
+                        colors: this.state.colors
+                    },
+                    type: "pie",
+                    values: this.state.values
+                } ],
+                height: this.props.height,
+                heightUnit: this.props.heightUnit,
+                layout: {
+                    autosize: this.props.responsive,
+                    showlegend: this.props.showLegend
+                },
+                style: PieChartContainer.parseStyle(this.props.style),
+                type: this.props.chartType,
+                width: this.props.width,
+                widthUnit: this.props.widthUnit
+            });
+        }
+
+        return null;
```

Comment:
What is the point of the default data you define in the PieChart component when u return null when there's no context object? Under what circumstances is the default data used?